middleware/grpc_instrumentation.go

```diff
@@ -1,23 +1,52 @@
 package middleware
 
 import (
+	"strconv"
 	"time"
 
 	"github.com/prometheus/client_golang/prometheus"
 	"golang.org/x/net/context"
+	spb "google.golang.org/genproto/googleapis/rpc/status"
 	"google.golang.org/grpc"
+	"google.golang.org/grpc/status"
 )
 
 // ServerInstrumentInterceptor instruments gRPC requests for errors and latency.
-func ServerInstrumentInterceptor(duration *prometheus.HistogramVec) grpc.UnaryServerInterceptor {
+func ServerInstrumentInterceptor(hist *prometheus.HistogramVec) grpc.UnaryServerInterceptor {
 	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
 		begin := time.Now()
 		resp, err := handler(ctx, req)
-		status := "success"
+		duration := time.Since(begin).Seconds()
+		respStatus := "success"
 		if err != nil {
-			status = "error"
+			errInfo, ok := status.FromError(err)
+			if ok {
+				respStatus = strconv.Itoa(int(errInfo.Code()))
```

Comment:
This could be rolled into the if, like this:
```go
if errInfo, ok := status.FromError(err); ok {
	respStatus = strconv.Itoa(int(errInfo.Code()))
}
```